### ISSUE TITLE
Configure Reset tool extents

### ISSUE THREAD

**Issue by jbednar** [holoviz/holoviews#4833]
Often when I'm working with a large Datashader dataset I want the initial extent of the plot to be zoomed in to show detail, but then (especially if using Box Zoom) it's very awkward to zoom out to see the big picture and select a different region to zoom into.

Can we add an option to configure the Bokeh Reset tool to have a different extent than the initial extent of the plot? That way I could set the initial extent as a like, configure the Reset tool to show the whole dataset, and then tell users to hit Reset if they want to choose a different region to zoom into.

**Comment by jlstevens** [comment]
An option like that seems like a reasonable request though it does break the expected semantics a bit. 'Reset' means 'set back to what it was' so having a reset different from the initialization would definitely be a weird default.

I suppose semantically this would be simulating several steps, namely initializing a plot with the resettable extents and immediately zooming into an ROI. What might be another idea would be a dedicated Bokeh tool specific for jumping to predefined ROIs e.g. a menu of (named) predefined viewports...

**Comment by jbednar** [comment]
Right; it sets back to some original configuration, in this case a configuration just before the actual plot shown, i.e. before limits were set on it.

Sure, such a tool would be useful, but it seems unlikely that enough people would configure it for users to learn about it, so it would probably sit there unused. Whereas people might be surprised when their Reset button shows the entire extent, but they will surely quickly realize what happened.

**Comment by jbednar** [comment]
This might also make sense as some sort of option for the Box Zoom tool, since it's the particular limitation of its one-way zoom that this addresses most directly. (Box zoom in, reset, box zoom in elsewhere, etc.)
